--- key-bindings.zsh	2022-04-25 10:19:45.881092078 +0200
+++ key-bindings.zsh	2022-04-25 10:19:26.441157873 +0200
@@ -39,6 +39,20 @@
 
 {
 
+fzf_binding() {
+  local -A default
+  default=(
+    file '^T'
+    cd '\ec'
+    history '^R'
+  )
+  local binding
+  if ! zstyle -s ':fzf:bindings' "$1" binding; then
+    binding="${default[$1]}"
+  fi
+  echo -n "$binding"
+}
+
 # CTRL-T - Paste the selected file path(s) into the command line
 __fsel() {
   setopt localoptions pipefail no_aliases 2> /dev/null
@@ -63,10 +77,10 @@
   return $ret
 }
 if [[ "${FZF_CTRL_T_COMMAND-x}" != "" ]]; then
-  zle     -N            fzf-file-widget
-  bindkey -M emacs '^T' fzf-file-widget
-  bindkey -M vicmd '^T' fzf-file-widget
-  bindkey -M viins '^T' fzf-file-widget
+  zle     -N                             fzf-file-widget
+  bindkey -M emacs "$(fzf_binding file)" fzf-file-widget
+  bindkey -M vicmd "$(fzf_binding file)" fzf-file-widget
+  bindkey -M viins "$(fzf_binding file)" fzf-file-widget
 fi
 
 # ALT-C - cd into the selected directory
@@ -86,10 +100,10 @@
   return $ret
 }
 if [[ "${FZF_ALT_C_COMMAND-x}" != "" ]]; then
-  zle     -N             fzf-cd-widget
-  bindkey -M emacs '\ec' fzf-cd-widget
-  bindkey -M vicmd '\ec' fzf-cd-widget
-  bindkey -M viins '\ec' fzf-cd-widget
+  zle     -N                           fzf-cd-widget
+  bindkey -M emacs "$(fzf_binding cd)" fzf-cd-widget
+  bindkey -M vicmd "$(fzf_binding cd)" fzf-cd-widget
+  bindkey -M viins "$(fzf_binding cd)" fzf-cd-widget
 fi
 
 # CTRL-R - Paste the selected command from history into the command line
@@ -110,10 +124,12 @@
   zle reset-prompt
   return $ret
 }
-zle     -N            fzf-history-widget
-bindkey -M emacs '^R' fzf-history-widget
-bindkey -M vicmd '^R' fzf-history-widget
-bindkey -M viins '^R' fzf-history-widget
+zle     -N                                fzf-history-widget
+bindkey -M emacs "$(fzf_binding history)" fzf-history-widget
+bindkey -M vicmd "$(fzf_binding history)" fzf-history-widget
+bindkey -M viins "$(fzf_binding history)" fzf-history-widget
+
+unset -f fzf_binding
 
 } always {
   eval $__fzf_key_bindings_options
